tests/virt_autotest/guest_upgrade_run.pm

```diff
@@ -108,15 +108,21 @@ sub run {
     my $self = shift;
     my $timeout = get_var('MAX_TEST_TIME', '36000') + 10;
     my $upload_log_name = 'guest-upgrade-logs';
-    script_run("echo \"Debug info: max_test_time is $timeout\"");
+    if (is_s390x) {
+        #ues die_on_timeout=> 0 as workaround for s390x test during call script_run, refer to poo#106765
+        script_run("echo \"Debug info: max_test_time is $timeout\"", die_on_timeout => 0);
+    } else {
+        script_run("echo \"Debug info: max_test_time is $timeout\"");
+    }
```

Comment:
Can here, both s390x and non-s390x use the same script_run with die_on_timeout =>0?

Reply:
@guoxuguang  I think you'd better raise this issue to openQA project or let them know. It seems that the change does not take this usage on s390x into consideration which means you have to use die_on_timeout => 0. Now you made the change which branches off s390x. I did not say branching off here is really bad here, but it does force us to make the same change to other arches here to some extent because it looks not very pleasant and leads that way.  Very interesting, we end up having to use die_on_timeout => 0 because of API change. I do not think this is what this API change initially would like to see.  
Maybe you use type_string here instead of script_run ? So we can avoid change to other tests ?

Reply:
Good point, use with `die_on_timeout =>0` as the simple way. Moreover, `die_on_timeout =>1` means that trigger command and finished this command in expected time. IMHO, use with `die_on_timeout =>1` is so useful for our osd test. But, find out this die_on_timeout =>1 do not work with s390x test. So, we'd better to keep with `die_on_timeout =>1` for non-s390x.

Reply:
I agree with Wayne. it is more acceptable that script_run() has different behavior between s390 with ipmi/qemu backend.

Reply:
yes, this PR just only the workaround for all s390x test. 

Also, open a ticket [poo#107164](https://progress.opensuse.org/issues/107164) to openQA tools team now.